vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/LinkManager.java

```diff
@@ -36,37 +38,66 @@ public abstract class LinkManager<L extends ILink<T>, T extends IVSpaceElement,
 
     @Autowired
     private IStorageEngine storage;
+    
+    @Autowired
+    private IImageService imageService;
+
+    protected abstract void deleteLinkRepo(L link);
+
+    protected abstract void deleteLinkDisplayRepo(L link);
+
+    protected abstract void removeFromLinkList(ISpace space, L link);
+
+    protected abstract U updateLinkAndDisplay(L link, U displayLink);
+
+    protected abstract U getDisplayLink(String linkDisplayId) throws LinkDoesNotExistsException;
+
+    protected abstract L getLink(String linkId);
 
+    protected abstract L createLinkObject(String title, String id);
+
+    protected abstract T getTarget(String linkedId);
+
+    protected abstract U createDisplayLink(L link);
+    
```

Comment:
I'm so confused, why is this here? I this needed for this ticket?

Reply:
ah I see, this should not be moved

Reply:
?

Reply:
This came from merging changes from the main branch